vortex-datafusion/src/lib.rs

```diff
@@ -261,7 +483,7 @@ impl ExecutionPlan for VortexMemoryExec {
             self.array.clone()
         };
 
-        Self::execute_single_chunk(chunk, &self.projection, context)
+        execute_unfiltered(&chunk, &self.scan_projection)
```

Comment:
not sure why this change, you immediately clone the chunk in the function. How about we also change the scan_projection to `&[usize]`?

Reply:
good point